pkg/destroy/aws/aws.go

```diff
@@ -185,19 +185,25 @@ func (o *ClusterUninstaller) RunWithContext(ctx context.Context) ([]string, erro
 	// running on the cluster creating new resources while we are attempting to delete resources, which could leak
 	// the new resources.
 	ec2Client := ec2.New(awsSession)
+	lastTerminateTime := time.Now()
 	err = wait.PollImmediateUntil(
 		time.Second*10,
 		func() (done bool, err error) {
-			instancesToDelete, err := o.findEC2Instances(ctx, ec2Client, deleted)
+			instancesRunning, instancesNotTerminated, err := o.findEC2Instances(ctx, ec2Client, deleted)
 			if err != nil {
 				o.Logger.WithError(err).Info("error while finding EC2 instances to delete")
 				if err := ctx.Err(); err != nil {
 					return false, err
 				}
 			}
-			if len(instancesToDelete) == 0 && err == nil {
+			if len(instancesNotTerminated) == 0 && len(instancesRunning) == 0 && err == nil {
 				return true, nil
 			}
+			instancesToDelete := instancesRunning
+			if time.Since(lastTerminateTime) > 9*time.Second {
```

Comment:
Do you have an idea what you want to use as the period here? 9 seconds is effectively "always".

Reply:
Hmm doesn't this wait 9 seconds until it requests for another fresh termination? Since the timeout to delete these running instances is 10 seconds, I was hoping to wait for 9 seconds before trying to request a fresh termination request.

Reply:
Yes, it waits 9 seconds. That wait time is much too short. It should be something like 10 minutes instead.

There is no "timeout to delete [those] running instances". There is a check on running instances every 10 seconds. We want to keep that check as is, but add a repeat of the terminate requests every 10 minutes for instances that are still shutting down.